packages/block-library/src/post-title/index.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Server-side rendering of the `core/post-title` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/post-title` block on the server.
+ *
+ * @return string Returns the filtered post title for the current post wrapped inside "h1" tags.
+ */
+function render_block_core_post_title() {
+	rewind_posts();
+	the_post();
+	return the_title( '<h1>', '</h1>', false );
```

Comment:
Shouldn't the heading level be variable? In a `singular` context, it should be `h1`. But on an archive context where there is more than one post in the loop, should it not be `h2`?

Reply:
Perhaps this should behave like the Heading block where you can pick from among the heading levels. There could be an “auto” setting which does this, but which you could also override as desired.

Reply:
>Perhaps this should behave like the Heading block where you can pick from among the heading levels. There could be an “auto” setting which does this, but which you could also override as desired.

Yeah I like that.

This was more of a first step to get something on the screen. We need a major design effort to come up with how these new blocks will behave. See the questions at the end of this overview issue: #17284